libs/memory/CMakeLists.txt

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) 2019 The STE||AR-Group
+#
+# SPDX-License-Identifier: BSL-1.0
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+cmake_minimum_required(VERSION 3.3.2 FATAL_ERROR)
+
+# Default location is $HPX_ROOT/libs/memory/include
+set(memory_headers
+  hpx/memory/intrusive_ptr.hpp
+  hpx/memory/detail/sp_convertible.hpp
+  hpx/memory/serialization/intrusive_ptr.hpp
+)
+
+# Default location is $HPX_ROOT/libs/memory/include_compatibility
+set(memory_compat_headers
+)
+
+# Default location is $HPX_ROOT/libs/memory/src
+set(memory_sources
+)
+
+include(HPX_AddModule)
+add_hpx_module(memory
+  COMPATIBILITY_HEADERS OFF
+  DEPRECATION_WARNINGS
+  FORCE_LINKING_GEN
+  GLOBAL_HEADER_GEN ON
+  EXCLUDE_FROM_GLOBAL_HEADER
+    "hpx/memory/detail/sp_convertible.hpp"
+    "hpx/memory/serialization/intrusive_ptr.hpp"
+  SOURCES ${memory_sources}
+  HEADERS ${memory_headers}
+  COMPAT_HEADERS ${memory_compat_headers}
+  DEPENDENCIES
+    hpx_config
+    hpx_assertion
+    hpx_serialization
```

Comment:
See my other comment. We should try to not depend on serialization here (although it's a much lighter dependency now!).